app/topics/content/content.component.ts

```diff
@@ -0,0 +1,224 @@
+    }
+  )
+}
+
+@Component({
+  selector: 'hip-content',
+  template: `
+  <div class="form" style="margin: 0;padding: 0;height: 100%;">
+    <div id="iframeEditor">
+    </div>
+  </div>
+`
+})
+export class ContentComponent implements OnInit {
+  private currentUser: User;
+  private topicId: number;
+
+  apiUrl = '';
+  public config = {
+    apiUrl: '',
+    editor: {
+      callbackUrl: '',
+      canBackToFolder: false,
+      curUserHostAddress: '',
+      documentType: '',
+      fileChoiceUrl: '',
+      firstName: '',
+      getServerUrl: '',
+      isEdit: false,
+      key: '',
+      lang: '',
+      lastName: '',
+      mode: '',
+      plugins: {},
+      type: '',
+      userid: '',
+    },
+    file: {
+      ext: '',
+      name: '',
+      uri: '',
+      version: 0,
+    },
+    history: [{}],
+    setHistoryData: {
+      url: '',
+      urlDiff: ''
+    }
+  };
+  loaded = true;
+  count = 0;
+
+  private docEditor: any = {
+    refreshHistory: function () {}
+  };
+
+  constructor(private route: ActivatedRoute,
+              private userService: UserService,
+              private toasterService: ToasterService,
+              private ooApiService: OOApiService,
+              private ngZone: NgZone) {
+    window.angularComponentRef = {
+      zone: this.ngZone,
+      component: this,
+      onRequestHistory: () => this.onRequestHistory(),
+      onRequestChangeHistoryData: (data: any) => this.onRequestChangeHistoryData(data)
+    };
+  }
+
+  ngOnInit() {
+    this.topicId = this.route.snapshot.params['id'];
+    this.ooApiService.getUrl('/topic/' + this.topicId + '/exists', {}).toPromise()
+      .then(
+        (response: any) => { this.getOnlyOffice(); }
+      )
+      .catch((error: any) => {
+        this.ooApiService.postUrl('/topic', '&topicId=' + this.topicId, {}).toPromise()
+          .then(
+            (response: any) => { this.getOnlyOffice(); }
+          );
+      });
+
+    // fetch current user and check permissions
+    this.userService.getCurrent().then(
+      (response: any) => {
+        this.currentUser = <User> response;
+      }
+    ).catch(
+      (error: any) => this.toasterService.pop('error', 'Error fetching current user', error)
+    );
+  }
+
+  ngOnDestroy() {
+    window.angularComponentRef = null;
+  }
+
+  onRequestHistory() {
+    let historyObj = this.config.history || null;
+
+    this.docEditor.refreshHistory(
+      {
+        currentVersion: this.config.file.version,
+        history: historyObj
+      });
+  }
+
+  onRequestChangeHistoryData(data: any) {
+    let version = data.data;
+    let url_arr = this.config.setHistoryData.url;
+    let urlDiff_arr = this.config.setHistoryData.urlDiff;
+
+    this.docEditor.setHistoryData({
+      version: version,
+      url: url_arr[version - 1] != "" ? url_arr[version - 1] : null,
+      urlDiff: urlDiff_arr[version - 1] != "" ? urlDiff_arr[version - 1] : null,
+    });
+  }
+
+  private getOnlyOffice() {
```

Comment:
as this function does not reall return anything, it shouldnt be named as if it was a getter (better: 'loadOnlyOffice')